Lib/fontTools/varLib/instancer.py

```diff
@@ -1284,6 +1298,158 @@ def instantiateVariableFont(
     return varfont
 
 
+def axisValuesFromAxisLimits(stat, axisLimits):
+
+    axisRecords = stat.table.DesignAxisRecord.Axis
+    axisValues = stat.table.AxisValueArray.AxisValue
+
+    format4 = [a for a in axisValues if a.Format == 4]
+    nonformat4  = [a for a in axisValues if a not in format4]
+    axisValues = format4 + nonformat4
+
+    axisOrder = {a.AxisOrdering: a.AxisTag for a in axisRecords}
+    pinnedAxes = set(k for k, v in axisLimits.items() if isinstance(v, (float, int)))
+
+    results, seen_axes = [], set()
+    for axisValue in axisValues:
+
+        # Ignore axisValue if it has ELIDABLE_AXIS_VALUE_NAME flag enabled.
+        # Enabling this flag will hide the axisValue in application font menus.
+        # TODO this is too greedy! we need to retain wght axisValues
+        if axisValue.Flags == 2:
+            continue
+
+        if axisValue.Format == 4:
+            axisIndexes = set(r.AxisIndex for r in axisValue.AxisValueRecord)
+            if seen_axes - axisIndexes != seen_axes:
+                continue
+            # TODO fix dup appends 
+            for rec in axisValue.AxisValueRecord:
+                axisTag = axisOrder[rec.AxisIndex]
+                if axisTag not in pinnedAxes:
+                    continue
+                if rec.Value == axisLimits[axisTag]:
+                    seen_axes.add(rec.AxisIndex)
+                    results.append((rec.AxisIndex, axisValue))
+
+        elif axisValue.Format in (1, 3):
+            axisTag = axisOrder[axisValue.AxisIndex]
+            # Add axisValue if it's used to link to another variable font
+            if axisTag not in axisLimits and axisValue.Value == 1.0:
+                seen_axes.add(rec.AxisIndex)
+                results.append((axisValue.AxisIndex, axisValue))
+
+            if axisTag not in pinnedAxes:
+                continue
+            # Add axisValue if its value is in the axisLimits and the user has
+            # pinned the axis
+            elif axisValue.Value == axisLimits[axisTag]:
+                seen_axes.add(rec.AxisIndex)
+                results.append((axisValue.AxisIndex,axisValue))
+
+        elif axisValue.Format == 2:
+            axisTag = axisOrder[axisValue.AxisIndex]
+            if axisTag not in pinnedAxes:
+                continue
+            if axisLimits[axisTag] >= axisValue.RangeMinValue \
+                and axisLimits[axisTag] <= axisValue.RangeMaxValue:
+                    seen_axes.add(axisValue.AxisIndex)
+                    results.append((axisValue.AxisIndex, axisValue))
+    return [v for k, v in sorted(results)]
```

Comment:
To be clear, this is sorting by `axisValue.AxisIndex`, which is an index into the `axisRecords` array; and not sorting by axis ordering values. Is that what's intended? (If so, probably worth calling out in a comment.)